lib/benchee/conversion/memory.ex

```diff
@@ -46,6 +46,43 @@ defmodule Benchee.Conversion.Memory do
               }
   }
 
+  # Unit conversion functions
+
+  @type unit :: Unit.t
+  @type unit_atom :: atom
```

Comment:
I like defining this type! However, I think we can be more specific. We only accept a very limited sub-set of atoms here, right? We're also allowed to use certain literals in type definitions, so maybe we can list the atoms exactly so the reader later on knows exactly which values are accepted? This way we can also lean on Dialyzer a bit to make sure we're not getting anything we don't want passed into this function.

So, for example, you could define this type like this:

```elixir
@type unit_atom :: :byte | :kilobyte | :megabyte | :gigabyte | :terabyte
```

Reply:
Nice! This is way more explicit, I somehow missed this capability when looking at the typespec docs

Reply:
@lwalter no worries, the typespec docs can be somewhat confusing, took me a while to get used to them :)